specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2023-07-01-preview/virtualMachineInstances.json

```diff
@@ -0,0 +1,1545 @@
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "VirtualMachineInstanceProperties": {
+      "type": "object",
+      "x-ms-client-flatten": true,
+      "properties": {
+        "hardwareProfile": {
+          "type": "object",
+          "properties": {
+            "vmSize": {
+              "type": "string",
+              "enum": [
+                "Default",
+                "Standard_A2_v2",
+                "Standard_A4_v2",
+                "Standard_D2s_v3",
+                "Standard_D4s_v3",
+                "Standard_D8s_v3",
+                "Standard_D16s_v3",
+                "Standard_D32s_v3",
+                "Standard_DS2_v2",
+                "Standard_DS3_v2",
+                "Standard_DS4_v2",
+                "Standard_DS5_v2",
+                "Standard_DS13_v2",
+                "Standard_K8S_v1",
+                "Standard_K8S2_v1",
+                "Standard_K8S3_v1",
+                "Standard_K8S4_v1",
+                "Standard_NK6",
+                "Standard_NK12",
+                "Standard_NV6",
+                "Standard_NV12",
+                "Standard_K8S5_v1",
+                "Custom"
+              ],
+              "default": "Default",
+              "x-ms-enum": {
+                "name": "VmSizeEnum",
+                "modelAsString": true
+              }
+            },
+            "processors": {
+              "type": "integer",
+              "format": "int32",
+              "description": "number of processors for the virtual machine instance"
+            },
+            "memoryMB": {
+              "type": "integer",
+              "format": "int64",
+              "description": "RAM in MB for the virtual machine instance"
+            },
+            "dynamicMemoryConfig": {
+              "type": "object",
+              "properties": {
+                "maximumMemoryMB": {
+                  "type": "integer",
+                  "format": "int64"
+                },
+                "minimumMemoryMB": {
+                  "type": "integer",
+                  "format": "int64"
+                },
+                "targetMemoryBuffer": {
+                  "type": "integer",
+                  "description": "Defines the amount of extra memory that should be reserved for a virtual machine instance at runtime, as a percentage of the total memory that the virtual machine instance is thought to need. This only applies to virtual systems with dynamic memory enabled. This property can be in the range of 5 to 2000.",
+                  "format": "int32"
+                }
+              }
+            }
+          },
+          "description": "HardwareProfile - Specifies the hardware settings for the virtual machine instance."
+        },
+        "networkProfile": {
+          "type": "object",
+          "properties": {
+            "networkInterfaces": {
+              "type": "array",
```

Comment:
>array [](http://example.com/codeflow?start=23&length=5)

use x-ms-identifiers for the object in the array: [https://github.com/Azure/autorest/tree/main/docs/extensions#x-ms-identifiers](https://github.com/Azure/autorest/tree/main/docs/extensions#x-ms-identifiers)

Reply:
That rule says "The default identifying property is id. You don't need to add this annotataion if there is the id property in the object model." The items in this array of networkInterfaces have a unique "id" property, so we don't need to add the x-ms-identifiers annotation, right?

Reply:
Yes, in this case you are good